### PR TITLE
All volumes manifest

### PR DESCRIPTION
**Pull request by sarastanton**
Create serializer, view, and url path for all volumes manifest

Addresses #810

### CODE REVIEW

**Comment by sarastanton** [comment]
Leaving this as a draft PR for now because I'm not quite sure it's working as intended, but it might make it easier to review/discuss here

**Comment by sarastanton** [comment]
@blms  I've made the corrections we discussed on Friday 👍

File: apps/iiif/serializers/all_volumes_manifest.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# pylint: disable = attribute-defined-outside-init, too-few-public-methods
+"""Module for serializing all volumes manifest"""
+from django.core.serializers.base import SerializerDoesNotExist
+import config.settings.local as settings
+from apps.iiif.serializers.base import Serializer as JSONSerializer
+
+class Serializer(JSONSerializer):
+    """IIIF All volumes manifest"""
```
Comment (blms) [inline apps/iiif/serializers/all_volumes_manifest.py:8]: I might change the docstring to something like `"""Serializer for a single IIIF manifest within the Readux all-volumes collection"""`

File: apps/iiif/manifests/views.py [hunk 2]
```diff
@@ -63,6 +63,38 @@ def get(self, request, *args, **kwargs): # pylint: disable = unused-argument
                 safe=False
             )
 
+class AllVolumesCollection(View):
+    """Endpoint for all volumes manifest."""
```
Comment (blms) [inline apps/iiif/manifests/views.py:67]: For consistency, maybe "all volumes collection" in this docstring

File: apps/iiif/manifests/views.py [hunk 3]
```diff
@@ -63,6 +63,38 @@ def get(self, request, *args, **kwargs): # pylint: disable = unused-argument
...
+
+        :return: All manifest objects
+        :rtype: django.db.models.QuerySet
+        """
+        return Manifest.objects.all()
+
+    def get(self, request, *args, **kwargs): # pylint: disable = unused-argument
+        """Responds to HTTP GET request for all manifests.
+
+        :return: IIIF representation of all volumes manifest
```
Comment (blms) [inline apps/iiif/manifests/views.py:79]: Same here, "return: IIIF representation of all volumes collection" in this docstring

**Review by blms — CHANGES_REQUESTED** [review]
Good progress! A few small changes and then I think we'll be done.

File: apps/iiif/serializers/all_volumes_manifest.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+            data = {
+                "@id": '{h}/iiif/v2/{p}/manifest'.format(h=settings.HOSTNAME, p=obj.pid),
+                "@type": "sc:Manifest",
+                "label": obj.label,
+                "modified": obj.updated_at.isoformat(),
+                "within": [{
+                    "@id": '{h}/iiif/v2/{p}/manifest'.format(h=settings.HOSTNAME, p=obj.pid),
+                    "@type": "sc:Manifest",
+                    "label": obj.label
+                } for collection in obj.collections.all()]
```
[the remark this answers is not in the document]
Comment (sarastanton) [reply apps/iiif/serializers/all_volumes_manifest.py]: 🤦‍♀️ You're absolutely right - I think we had even talked about this one. Sorry! Updated now 👍

File: apps/iiif/manifests/views.py (hunk 2 above)
Comment (sarastanton) [reply apps/iiif/manifests/views.py]: Updated as suggested

File: apps/iiif/manifests/views.py (hunk 3 above)
Comment (sarastanton) [reply apps/iiif/manifests/views.py]: Updated as suggested

File: apps/iiif/serializers/all_volumes_manifest.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+            return data
+        return None
+
+class Deserializer:
+    """Deserialize IIIF all volumes manifest
+
+    :raises SerializerDoesNotExist: Not yet implemented.
+    """
+    def __init__(self, *args, **kwargs):
+        raise SerializerDoesNotExist("all_vall_volumes_manifest is a serialization-only serializer")
```
Comment (blms) [inline apps/iiif/serializers/all_volumes_manifest.py:32]: Just noticed the minor typo here!
```suggestion
        raise SerializerDoesNotExist("all_volumes_manifest is a serialization-only serializer")
```

**Review by blms — APPROVED** [review]
Great work! Can merge once the url is changed to `iiif/v2/all-volumes-collection` and the typo is corrected.

File: apps/iiif/serializers/all_volumes_manifest.py (hunk 5 above)
Comment (sarastanton) [reply apps/iiif/serializers/all_volumes_manifest.py]: 🤦‍♀️ Good catch - fixed!

**Comment by jayvarner** [comment]
Thanks for all the work!

**Pull request merged**